annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,145 @@
+package edu.isi.vista.annotationutils
+
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.Session
+import edu.isi.nlp.parameters.serifstyle.SerifStyleParameterFileLoader
+import java.io.File
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.ResetCommand
+import org.eclipse.jgit.api.TransportConfigCallback
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.transport.*
+import org.eclipse.jgit.treewalk.CanonicalTreeParser
+import org.eclipse.jgit.util.FS
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * The goal of this program is to export the annotations, save the restored
+ * documents, compare with the previous commit to see if any changes have been
+ * made, and if so, push those changes to the repo.
+ *
+ * The program takes one argument: a file containing these parameter values:
+ *
+ * <ul>
+ *     <li> {@code exportAnnotationsParams}: the location of the parameter file meant for {@code ExportAnnotations.kt} </li>
+ *     <li> {@code restoreJsonParams}: the location of the parameter file meant for {@code RestoreJson.kt} </li>
+ *     <li> {@code repoPath}: the location of the local repository
+ *     (https://github.com/isi-vista/curated-training-annotation)</li>
+ * </ul>
+ *
+ * This program currently assumes these things:
+ * <ul>
+ *     <li>The annotations will be saved to .../curated-training-annotation/data/exported</li>
+ *     <li>Your ssh private key is saved in ~/.ssh/id_rsa.</li>
+ * </ul>
+ */
+
+
+fun pushNew(repoPath: String) {
+
+    // Source: https://github.com/kittinunf/Kotlin-Playground/blob/master/buildSrc/src/main/java/com/github/kittinunf/gradle/ReleaseToMasterMergeTask.kt
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val git = Git.open(File(repoPath))
```

Comment:
So, up to this point in reading this, I had assumed that `repoPath` was the URL of the remote repository you want to push to, but now it appears to be the path to the local working copy of that repository.  We should clarify this in the documentation.

However, in the interest of being self-contained, I'd suggest making a slight change to the design:
* have two parameters `localWorkingCopyDirectory` and `repositoryToPushTo`.
* if `localWorkingCopyDirectory` does not exist, clone `repositoryToPushTo` to that location.
* if it does exist, make sure it is in the expected state
     * check that its remote repository URL matches `repositoryToPushTo`.
     * check if has no local modifications (`status.isClean())`
     * check it is on the master branch
     * sync it up to the remote master

Reply:
Done; wrote new method `setUpRepository` to perform actions depending on existence of `localWorkingCopyDirectory`